src/components/app/content/card/index.js

```diff
@@ -0,0 +1,47 @@
+import React, { useState } from 'react';
+import './index.css';
+import { updateCard } from '../../../../redux/actions';
+import { connect } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+
+function Card(props) {
+    const [tempState, setTempState] = useState(props.location.cardData);
+    const history = useHistory();
+
+    const fillData = (data) => setTempState({ ...tempState, ...data });
+
+    const saveCardHandler = () => {
+        props.updateCard(tempState);
+        history.push('/');
```

Comment:
Why return to list on each save? No need. Should be fixed by using alreasdy created Card component, save will switch card to view mode..